dali/kernels/imgproc/paste/paste_gpu.h

```diff
@@ -174,8 +175,11 @@ void CreateSampleDescriptors(
           patch.patch_end[1] = scaled_x_to_x[x + 1];
           patch.in = nullptr;  // to be filled later
           patch.in_sample_idx = max_paste == -1 ? -1 : sample.inputs[max_paste].in_idx;
+          patch.in_batch_idx = max_paste == -1 ? -1 : sample.inputs[max_paste].batch_idx;
           patch.in_pitch[0] = 0;
-          patch.in_pitch[1] = max_paste == -1 ? -1 : in_shape[sample.inputs[max_paste].in_idx][1];
+          int mp_in_idx = sample.inputs[max_paste].in_idx;
+          int mp_batch_idx = sample.inputs[max_paste].batch_idx;
+          patch.in_pitch[1] = max_paste == -1 ? -1 : in_shapes[mp_batch_idx][mp_in_idx][1];
```

Comment:
Btw, what kind of pitch is that and why do we need to take it from the last pasted sample as far as I understand?

Reply:
My understanding is that we split the image into non-overlapping recatngular patches, where each patch has a single source and for each sample we take the top-most sample, the pithc is the row stride. At least from a glimpse into the code.